packages/node_modules/@node-red/nodes/core/network/21-httprequest.js

```diff
@@ -205,23 +205,25 @@ module.exports = function(RED) {
                     }
                 }
             }
-            if (this.credentials) {
+            var parsedURL = new URL(url)
```

Comment:
```suggestion
            this.credentials = this.credentials || {}
            var parsedURL = new URL(url)
```

Reply:
Not sure we need this one, but better safe than sorry

Reply:
But adding it does explain the earlier test failure, (I think when run for real `this.credentials` is always `{}` but it's just not added in the tests)